### ISSUE TITLE
I1022 raised when converting List<String> to String via Fn::Join

### ISSUE THREAD

**Issue by verbitan** [aws-cloudformation/cfn-lint#3682]
### CloudFormation Lint Version

cfn-lint 1.12.4

### What operating system are you using?

Red Hat Enterprise Linux release 8.10 (Ootpa)

### Describe the bug

The reproduction template raises the error `I1022: Prefer using Fn::Sub over Fn::Join with an empty delimiter`.

### Expected behavior

I don't think there's a suitable alternative to the reproduction template. If you switch to using `Fn::Sub` instead then the error `E8003: 'PermittedCIDRs' is not of type 'string'` is raised. I also don't believe it's possible to suppress this check inline within a `Conditions:` block.

### Reproduction template

```yaml
AWSTemplateFormatVersion: "2010-09-09"

Description: Example I1022 failure.

Parameters:
  PermittedCIDRs:
    Description: CIDR ranges permitted.
    Type: List<String>

Conditions:
  AllowEverything:
    Fn::Equals:
      - Fn::Join:
        - ""
        - !Ref PermittedCIDRs
      - "0.0.0.0/0"
```

**Comment by kddejong** [comment]
This is an interesting scenario for this use case though.  I think you have it correct but debating if this should be an exception to `I1022`
